File: server/job/src/main/java/io/iamcyw/ams/job/strategy/persistence/AlarmSourcePO.java
package io.iamcyw.ams.job.strategy.persistence;

import io.quarkus.hibernate.orm.panache.PanacheEntity;

import javax.persistence.Column;
import javax.persistence.Entity;

@Entity(name = "ALARM_SOURCE")
public class AlarmSourcePO extends PanacheEntity {

    @Column(name = "name")
    public String name;

    public static AlarmSourcePO findByName(String name) {
        return find("name", name).singleResult();
    }

}

File: domain/src/jvmMain/kotlin/io/iamcyw/ams/domain/job/strategy/AlarmSourcePO.kt
package io.iamcyw.ams.domain.job.strategy

import io.quarkus.hibernate.orm.panache.kotlin.PanacheCompanion
import io.quarkus.hibernate.orm.panache.kotlin.PanacheEntityBase
import javax.persistence.Column
import javax.persistence.Entity
import javax.persistence.GeneratedValue
import javax.persistence.Id

@Entity(name = "ALARM_SOURCE")
@kotlinx.serialization.Serializable
actual open class AlarmSourcePO actual constructor() : PanacheEntityBase {

    companion object : PanacheCompanion<AlarmSourcePO> {
        fun findByName(name: String) = find("name", name).singleResult()

    }

    @field:Id
    @field:GeneratedValue
    actual open var id: Long? = null

    @get:Column(name = "name")
    actual open var name: String = ""

    actual constructor(id: Long) : this() {
        this.id = id
    }

    actual constructor(id: Long, name: String) : this() {
        this.id = id
        this.name = name
    }


}